sdk/cosmosdb/cosmos/src/encryption/Cache/ClientEncryptionKeyPropertiesCache.ts

```diff
@@ -5,15 +5,25 @@ import { ClientEncryptionKeyProperties } from "../ClientEncryptionKey";
 
 /**
  * The cache used to store the properties of the client encryption key
+ * @hidden
  */
+
 export class ClientEncryptionKeyPropertiesCache {
+  private static instance: ClientEncryptionKeyPropertiesCache;
   // key is database id + '/'+ clientEncryptionKeyId
   private clientEncryptionKeyPropertiesCache: Map<string, ClientEncryptionKeyProperties>;
 
-  constructor() {
+  private constructor() {
     this.clientEncryptionKeyPropertiesCache = new Map<string, ClientEncryptionKeyProperties>();
   }
 
+  public static getInstance(): ClientEncryptionKeyPropertiesCache {
+    if (!ClientEncryptionKeyPropertiesCache.instance) {
+      ClientEncryptionKeyPropertiesCache.instance = new ClientEncryptionKeyPropertiesCache();
+    }
+    return ClientEncryptionKeyPropertiesCache.instance;
+  }
+
   public getClientEncryptionKeyProperties(key: string): ClientEncryptionKeyProperties | undefined {
```

Comment:
add logic for `forceRefresh`. If this flag is passed, latest value is fetched from the backend even if value is present in the cache.